projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php

```diff
@@ -628,6 +628,19 @@ public function handle_preferred_view() {
 			return;
 		}
 		$this->set_preferred_view( $this->get_current_screen(), $_GET['preferred-view'] );
+
+		/**
+		 * Dashboard Quick switcher action triggered when a user switches to a different view.
+		 *
+		 * @module masterbar
+		 *
+		 * @since 9.9.1
+		 *
+		 * @param string The current screen of the user.
+		 * @param string The preferred view the user selected.
+		 */
+		\do_action( 'jetpack_dashboard_switcher_changed_view', $this->get_current_screen(), $_GET['preferred-view'] );
```

Comment:
Nitpick: Now that we use `$this->get_current_screen()` and `$_GET['preferred-view']` in another place, I wonder if it's a good time to create vars to make the code more legible:

```php
$preferred_view = isset( $_GET['preferred-view'] ? $_GET['preferred-view'] : null;
$current_screen = $this->get_current_screen();

if ( in_array( $preferred_view, array( self::DEFAULT_VIEW, self::CLASSIC_VIEW ), true ) ) {
	return;
}

$this->set_preferred_view( $current_screen, $preferred_view );

\do_action( 'jetpack_dashboard_switcher_changed_view', $current_screen, $preferred_view );

```

Reply:
I guess we can clean it a bit. :) I wanted to have a smaller diff in there so that it's easier to review.